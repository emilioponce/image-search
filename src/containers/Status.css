.Status {
  color: #5fd6f6;
}
