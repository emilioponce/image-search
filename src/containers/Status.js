import { connect } from 'react-redux'
import React from 'react'
import { getTotalAmountOfResultsSelector } from '../reducers/selectors'
import { BIG_AMOUNT_OF_RESULTS } from '../config/constants'
import PropTypes from 'prop-types'

import './Status.css'

const Status = ({ total }) => {
  if (!total) {
    return null
  }

  const text =
    total > BIG_AMOUNT_OF_RESULTS ? `Large amount of results` : `Results found`
  return <div className="Status">{`${text} (${total} images)`}</div>
}

const mapStateToProps = state => {
  return {
    total: getTotalAmountOfResultsSelector(state)
  }
}

Status.propTypes = {
  total: PropTypes.string
}

export default connect(
  mapStateToProps,
  null
)(Status)
