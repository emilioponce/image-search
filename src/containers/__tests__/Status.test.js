import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Status } from '../Status'

configure({ adapter: new Adapter() })

const component = <Status total={'1'} />

describe('Status component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(component)
  })

  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matches', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should show an exact text for less than a big amount of results', () => {
    const component = <Status total={'100'} />
    const wrapper = shallow(component)
    expect(wrapper.find('.Status').text()).toBe('Results found (100 images)')
  })

  it('Should show an exact text for a big amount of results', () => {
    const component = <Status total={'1001'} />
    const wrapper = shallow(component)
    expect(wrapper.find('.Status').text()).toBe(
      'Large amount of results (1001 images)'
    )
  })
})
