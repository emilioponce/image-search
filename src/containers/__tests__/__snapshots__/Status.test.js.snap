// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component Snapshot matches 1`] = `
<div
  className="Status"
>
  Results found (1 images)
</div>
`;
